Volume/Views/MainView/HomeView.swift

```diff
@@ -0,0 +1,231 @@
+        // Invisible navigation link
+        // Only opens if application is opened through deeplink w/ valid article
+        Group {
+            if let articleID = viewModel.deeplinkID {
+                NavigationLink("", destination: BrowserView(initType: .fetchRequired(articleID), navigationSource: .morePublications), isActive: $viewModel.openArticleFromDeeplink)
+            }
+        }
+    }
+
+    // MARK: Sections
+
+    private var trendingArticlesSection: some View {
+        Section {
+            ScrollView(.horizontal, showsIndicators: false) {
+                switch viewModel.trendingArticles {
+                case .loading, .reloading:
+                    HStack(spacing: Constants.trendingArticleHorizontalSpacing) {
+                        ForEach(0..<2) { _ in
+                            BigReadArticleRow.Skeleton()
+                        }
+                    }
+                case .results(let articles):
+                    HStack(spacing: Constants.trendingArticleHorizontalSpacing) {
+                        ForEach(articles) { article in
+                            NavigationLink {
+                                BrowserView(initType: .readyForDisplay(article), navigationSource: .trendingArticles)
+                            } label: {
+                                BigReadArticleRow(article: article)
+                            }
+                        }
+                    }
+                }
+            }
+        } header: {
+            Header("The Big Read")
+                .padding(.vertical, Constants.rowVerticalPadding)
+                .foregroundColor(.black)
+        }
+        .background(headerGradient)
+    }
+
+    private var weeklyDebriefButton: some View {
+        Group {
+            switch viewModel.weeklyDebrief {
+            case .loading, .reloading:
+                SkeletonView()
+            case .results(let weeklyDebrief):
+                if let _ = weeklyDebrief {
+                    WeeklyDebriefButton(buttonPressed: $viewModel.isWeeklyDebriefOpen)
+                }
+            }
+        }
+        .padding(.top, Constants.weeklyDebriefTopPadding)
+        .padding(.bottom, Constants.rowVerticalPadding)
+    }
+
+    private var weeklyDebriefView: some View {
+        Group {
+            if let weeklyDebrief = userData.weeklyDebrief {
+                WeeklyDebriefView(
+                    isOpen: $viewModel.isWeeklyDebriefOpen,
+                    onOpenArticleUrl: $viewModel.deeplinkID,
+                    openedURL: $viewModel.openArticleFromDeeplink,
+                    weeklyDebrief: weeklyDebrief)
+            }
+        }
+    }
+
+    private var followedArticlesSection: some View {
+        articleSection(followed: true)
+    }
+
+    private var unfollowedArticlesSection: some View {
+        articleSection(followed: false)
+    }
+
+    private func articleSection(followed: Bool) -> some View {
+        Section {
```

Comment:
I feel like this could be written to more cleanly deal with `followed`, so you don't need so many conditionals, but idk that may be hard

Reply:
Meh I think there are few enough conditionals (only 1 for each `switch case`) that this is as simple as we can readably write.